### PR TITLE
Change `PresignParticipant` to use new `LocalStorage` API

### LINKED ISSUE #185: Change `PresignParticipant` to use our type-based local storage.
- [x] Change `PresignParticipant`'s `storage: Storage` field to use the new `LocalStorage` type as defined in #181.
- [x] Update all calls `PresignParticipant` makes to its `storage` field to use the new API.
- [x] As this is the last protocol in which we'll swap in the type-based local storage, replace `process_ready_message` with the implementation in `process_ready_message_local` provided in #206.

### PR DESCRIPTION
**Pull request by amaloz**
Closes #185.

### CODE REVIEW

File: src/protocol.rs [hunk 1]
```diff
@@ -289,16 +286,20 @@ impl Participant {
...
 
         // Clear the presign record after being used once
-        let _ =
-            self.main_storage
-                .delete(PresignStorageType::Record, presign_identifier, self.id)?;
+        let _ = self.main_storage.delete(
+            PersistentStorageType::PresignRecord,
+            presign_identifier,
+            self.id,
+        )?;
```
Comment (marsella) [inline src/protocol.rs:302]: thought: this is not really the point of this PR but I would be inclined to delete this even if `sign` fails on L294, e.g. immediately after `retrieve`.

File: src/presign/participant.rs [hunk 2]
```diff
@@ -44,36 +44,60 @@ use crate::{
...
+#[derive(Debug)]
 pub(crate) struct PresignParticipant {
     /// A unique identifier for this participant
     id: ParticipantIdentifier,
     /// A list of all other participant identifiers participating in the
     /// protocol
     other_participant_ids: Vec<ParticipantIdentifier>,
+    /// Old storage mechanism currently used to store persistent data
+    ///
+    /// TODO #205: To be removed once we remove the need for persistent storage
```
Comment (marsella) [inline src/presign/participant.rs:97]: nit: This should reference #180, not #205. 180 removes main storage from `ProtocolParticipant`s. 205 removes it from `Participant`.

File: src/participant.rs [hunk 3]
```diff
@@ -165,58 +165,12 @@ pub(crate) trait ProtocolParticipant {
...
-    /// once we've transitioned all of the subprotocols to use
-    /// `LocalStorage`.
-    fn process_ready_message_local<T: TypeTag<Value = ()>>(
+    fn process_ready_message<T: TypeTag<Value = ()>>(
         &self,
         message: &Message,
         storage: &LocalStorage,
     ) -> Result<(ProcessOutcome<Self::Output>, bool)> {
-        // TODO #185: Unlike in `process_ready_message`, we don't store the ready
+        // TODO #205: Unlike before, we don't store the ready
```
Comment (marsella) [inline src/participant.rs:173]: nit: This should reference 180, not 205.

**Review by marsella — APPROVED** [review]
Looks good to me, just two tiny docs nits.

File: src/presign/participant.rs (hunk 2 above)
Comment (amaloz) [reply src/presign/participant.rs]: Yes good catch! I updated the other references to 205 as well.

**Pull request merged**